CMakeLists.txt

```diff
@@ -207,6 +209,13 @@ foreach(s ${sources})
 	list(APPEND sources2 src/${s})
 endforeach(s)
 
+if (shared)
+	list(APPEND sources2 ${asio_sources})
+	add_definitions(-DBOOST_SYSTEM_DYN_LINK -DBOOST_CHRONO_DYN_LINK -DBOOST_RANDOM_DYN_LINK)
+else()
+	message(STATUS "When linking static libtorrent, define BOOST_ASIO_SEPARATE_COMPILATION and include boost/asio/impl(/ssl)/src.hpp")
```

Comment:
It could be done, but the point is that it shouldn't. I'll explain in a comment at the pull request top level why this responsibility should remain with the user of the library.